### PR TITLE
Allow specifying additional info on call to profile

### PR DESCRIPTION
**Pull request by matklad**
bors r+
🤖

### CODE REVIEW

File: crates/ra_hir_ty/src/db.rs [hunk 1]
```diff
@@ -111,7 +112,15 @@ pub trait HirDatabase: DefDatabase {
 }
 
 fn infer(db: &impl HirDatabase, def: DefWithBodyId) -> Arc<InferenceResult> {
-    let _p = profile("wait_infer");
+    let _p = profile("wait_infer").detail(|| match def {
+        DefWithBodyId::FunctionId(it) => db.function_data(it).name.to_string(),
```
Comment (flodiebold) [inline crates/ra_hir_ty/src/db.rs]: it would be nicer to do this without causing salsa deps, right?

Comment (matklad) [reply crates/ra_hir_ty/src/db.rs]: Yup, but this won't trigger unless profiling is enabled, so I think it's ok enough

Comment (flodiebold) [reply crates/ra_hir_ty/src/db.rs]: ... but I don't know if there's even a way to do that

Comment (matklad) [reply crates/ra_hir_ty/src/db.rs]: Profiling is controlled by `RA_PROFILE` environment variable

**Pull request merged**